portfolio/src/main/webapp/script.js

```diff
@@ -48,12 +56,52 @@ function getMessages() {
   });
 }
 
+
+
+//check whether user is logged in using async
+async function checkWhetherUserIsLoggedIn() {
+  //set visibility of form and logurl to none
+  document.getElementById('comment-form').style.display="none";
+  document.getElementById('logInOut').style.display="none";
+  
+  //get login status
+  const response = await fetch('/login');
+  const quote = await response.text();
+  console.log(quote);
+
+  //convert login information to json object
+  var obj = JSON.parse(quote);
+  console.log(obj.isLoggedIn);
+  console.log(obj.logUrl);
+
+  //display form and logUrl based on whether user is logged in or not
+  if(obj.isLoggedIn === "true"){
+      document.getElementById('comment-form').style.display="block";
+      document.getElementById('logInOut').innerHTML = '';
```

Comment:
You get this same element a few different times. You might want to store it in a variable instead, that way you only have to get it once?